packages/pubsub/src/index.js

```diff
@@ -553,8 +553,15 @@ PubSub.prototype.getTopicsStream = common.paginator.streamify('getTopics');
  *     messages. (default: 10)
  * @param {number} options.maxInProgress - Maximum messages to consume
  *     simultaneously.
+ * @param {number|date} options.messageRetentionDuration - Set this to override
+ *     the default duration of 7 days. This value is expected in seconds (length
+ *     of retention) or a Date (expiration of retention). Acceptable values are
```

Comment:
Thanks for helping me understand. I will remove this functionality and only accept a number to represent seconds.